lms/templates/dashboard/credit_info.underscore

```diff
@@ -0,0 +1,86 @@
+            credit_msg = _.sprintf(gettext("Thank you for your payment. To receive course credit, you must now request credit at the %(link_to_provider_site)s  website. Select <b>Request Credit</b> to get started."), {
+                 link_to_provider_site: provider_link
+            });
+            credit_msg_class = "credit-request-not-started-msg";
+            credit_btn_label = gettext("Request Credit");
+            credit_btn_class = 'pending-credit-btn';
+        } else if (request_status == 'pending') {
+            //# Request received but not reviewed
+            //## Translators: provider_name is the name of a credit provider or university (e.g. State University)
+            credit_msg = _.sprintf(gettext("%(provider_name)s has received your course credit request. We will update you when credit processing is complete."),
+            {
+                provider_name: model.credit_status.provider_name
+            });
+
+            credit_msg_class = "credit-request-pending-msg";
+            credit_btn_label = gettext("View Details");
+            credit_btn_class = 'pending-credit-btn';
+        } else if (request_status == 'approved') {
+            //# Credit granted!
+            //# Translators: link_to_provider_site is a link to an external webpage. The text of the link will be the name of a credit provider, such as 'State University' or 'Happy Fun Company'. provider_name is the name of credit provider.
+            credit_msg = _.sprintf(gettext("<b>Congratulations!</b> %(provider_name)s has approved your request for course credit. To see your course credit, visit the %(link_to_provider_site)s website."),
+            {
+                provider_name:model.credit_status.provider_name,
+                link_to_provider_site: provider_link
+            });
+            credit_msg_class = "credit-request-approved-msg";
+            credit_btn_href = model.credit_status.provider_status_url;
+            credit_btn_label = gettext("View Credit");
+        } else if (request_status == 'rejected') {
+            //# REJECTED (by the credit provider)!
+            //## Translators: link_to_provider_site is a link to an external webpage. The text of the link will be the name of a credit provider, such as 'State University' or 'Happy Fun Company'. provider_name is the name of credit provider.
+            credit_msg= _.sprintf(gettext("%(provider_name)s did not approve your request for course credit. For more information, contact %(link_to_provider_site)s directly."),
+            {
+                provider_name: model.credit_status.provider_name,
+                link_to_provider_site: provider_link,
+            });
+            credit_msg_class = "credit-request-rejected-msg";
+            credit_btn_label = null;
+        }
+    }
+%>
```

Comment:
All of the code up to this point should be moved into the View. This isn't rendering anything onto the page, just formatting data. The View is where the data should be formatted.